File: context_tools/stories/templates/js/{epic-verb-noun}/{sub-epic-verb-noun}/{story-verb-noun}/{story_verb_noun}_test_helper.domain.js
/**
 * ---
 * fidelity: [engineering]
 * artifact: [test-helper]
 * format: js
 * ---
 *
 * Tier: domain - {StoryVerbNoun}Helper backed by direct domain-class calls.
 * Real: domain-core class. Stubbed: nothing.
 */

import { describe } from "node:test";
import { create{StoryVerbNoun}Story } from "./{story_verb_noun}_story.js";

class DomainHelper {
  givenPrecondition() {
    throw new Error("not implemented: givenPrecondition");
  }
  whenAction() {
    throw new Error("not implemented: whenAction");
  }
  thenOutcome() {
    throw new Error("not implemented: thenOutcome");
  }
}

describe("tier: domain", () => {
  create{StoryVerbNoun}Story(new DomainHelper());
});

File: context_tools/stories/templates/py/{epic-verb-noun}/{sub-epic-verb-noun}/{story-verb-noun}/{story_verb_noun}_test_helper.domain.py
# ---
# fidelity: [engineering]
# artifact: [test-helper]
# format: py
# ---
#
# Tier: domain - {EpicVerbNoun}Helper backed by direct domain-class calls.
# Real: domain-core class. Stubbed: nothing.

"""Domain tier test-helper for `{Story Name}`."""

from __future__ import annotations

from {story_verb_noun}_story import create_{story_verb_noun}_story


class DomainHelper:
    def given_precondition(self) -> None:
        raise NotImplementedError("not implemented: given_precondition")

    def when_action(self) -> None:
        raise NotImplementedError("not implemented: when_action")

    def then_outcome(self) -> None:
        raise NotImplementedError("not implemented: then_outcome")


globals().update(create_{story_verb_noun}_story(DomainHelper()))
